pkg/cli/dev.go

```
@@ -59,7 +68,76 @@ func (s *Dev) Run(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	imageSource := imagesource.NewImageSource(s.client.AcornConfigFile(), s.File, s.ArgsFile, args, nil, z.Dereference(s.AutoUpgrade))
+	var imageSource imagesource.ImageSource
+	if s.Clone == "" {
+		imageSource = imagesource.NewImageSource(s.client.AcornConfigFile(), s.File, s.ArgsFile, args, nil, z.Dereference(s.AutoUpgrade))
+	} else {
+		// Get info from the running app
+		app, err := c.AppGet(cmd.Context(), s.Clone)
+		if err != nil {
+			return err
+		}
+		vcs := app.Status.Staged.AppImage.VCS
+
+		if len(vcs.Remotes) == 0 {
+			return fmt.Errorf("clone can only be done on an app built from a git repository")
+		}
+
+		for _, remote := range vcs.Remotes {
+			var gitUrl string
+			httpUrl, err := url.Parse(remote)
+			if err == nil {
+				gitUrl = fmt.Sprintf("git@%s:%s", httpUrl.Host, httpUrl.Path[1:])
+			} else {
+				gitUrl = remote
+			}
+
+			// TODO workdir named after git repo, cloned app name, or just this app's name?
+			idx := strings.LastIndex(gitUrl, "/")
+			if idx < 0 || idx >= len(gitUrl) {
+				fmt.Printf("failed to determine repository name %q\n", gitUrl)
+				continue
+			}
+			workdir := strings.TrimSuffix(gitUrl[idx+1:], ".git")
+
+			// Clone git repo
+			auth, _ := ssh.NewSSHAgentAuth("git")
```

Why are we ignoring the error from `NewSSHAgentAuth`?